src/openforms/forms/tests/admin/test_form.py

```diff
@@ -550,6 +550,26 @@ def test_form_admin_copy(self):
             copied_form.confirmation_email_template.id, confirmation_email_template.id
         )
 
+    @tag("gh-3378")
+    def test_copy_form_with_trigger_from_step_in_logic(self):
+        user = UserFactory.create(is_superuser=True, is_staff=True)
+        self.client.force_login(user)
+
+        form = FormFactory.create(
+            authentication_backends=["digid"], internal_name="internal"
+        )
+        admin_url = reverse("admin:forms_form_change", args=(form.pk,))
+        form_step = FormStepFactory.create(form=form, form_definition__is_reusable=True)
+        FormLogicFactory.create(form=form, trigger_from_step=form_step)
+
+        # React UI renders this input, so simulate it in a raw POST call
+        self.client.post(admin_url, data={"_copy": "Copy"})
+
+        copied_form = Form.objects.get(slug=_("{slug}-copy").format(slug=form.slug))
+        copied_logic = copied_form.formlogic_set.get()
+
+        self.assertEqual(copied_logic.trigger_from_step, copied_form.formstep_set.get())
```

Comment:
I think we can simplify this test case by only keeping what's relevant for this particular issue:

```suggestion
    @tag("gh-3378")
    def test_copy_form_with_trigger_from_step_in_logic(self):
        user = SuperUserFactory.create()
        self.client.force_login(user)

        form_step = FormStepFactory.create()
        FormLogicFactory.create(form=form_step.form, trigger_from_step=form_step)
        admin_url = reverse("admin:forms_form_change", args=(form_step.form.pk,))

        # React UI renders this input, so simulate it in a raw POST call
        self.client.post(admin_url, data={"_copy": "Copy"})

        copied_form = Form.objects.exclude(pk=form_step.form.pk).get()
        copied_step = FormStep.objects.exclude(pk=form_step.pk).get()
        copied_logic = copied_form.formlogic_set.get()

        self.assertEqual(copied_logic.trigger_from_step, copied_step)
```

Authentication backends or any form-specific configuration is completely irrelevant here - and we can explicitly query for the copied objects (form, step, logic) since the test data is set up so that there's only 1 form to begin with and after the copy there should only be one additional form. Using the `queryset.get()` approach enforces this and prevents surprises/mistakes in test data setup.